File: Programme_Python/signaux/exo-2-saisit.py
import os
import sys
import signal

if len(sys.argv) != 2:
    print(f"Usage: {sys.argv[0]} <PID_affiche>")
    sys.exit(1)

pid_affiche = int(sys.argv[1])

# Saisie de l'entier
entier = int(input("Saisir un entier : "))

# Ecriture dans le fichier
try:
    with open("/tmp/entier.txt", "w") as f:
        f.write(f"{entier}\n")
except IOError as e:
    print("Erreur lors de l'écriture :", e)
    sys.exit(1)

# Envoi du signal au processus affiche
try:
    os.kill(pid_affiche, signal.SIGUSR1)
    print(f"Signal envoye au processus {pid_affiche}")
except OSError as e:
    print("Erreur lors de l'envoi du signal :", e)
    sys.exit(1)

File: Programme_C/chapitre-7/exo-2-saisit.c
/*Ecrire un programme saisit.c qui saisit un int au clavier, et l’enregistre
dans un fichier /tmp/entier.txt. Écrire un programme affiche.c qui attend (avec sleep) un
signal utilisateur du programme saisit.c. Lorsque l’entier a été saisi, le programme affiche.c
affiche la valeur de l’entier.*/

#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <unistd.h>

int main(int argc, char *argv[])
{
    if (argc != 2) 
     {
        fprintf(stderr, "Usage: %s <PID_affiche>\n", argv[0]);
        return 1;
     }

    pid_t pid_affiche = atoi(argv[1]);
    int entier = 0 ;

    printf("Saisir un entier : ");
    scanf("%d", &entier);

   
    FILE *fichier = fopen("/tmp/entier.txt", "w");

    if (!fichier) 
     {
        perror("fopen");
        return 1;
     }

    fprintf(fichier, "%d\n", entier);
    fclose(fichier);

    if (kill(pid_affiche, SIGUSR1) == -1) 
    {
        perror("kill() :");
        return 1;
    }

    printf("Signal envoye au processus %d\n", pid_affiche);
    return 0;
}